Add page and page_size to find query

mandruis7 opened the pull request:
This is in order to go over the 20 objects that We've with the default limit in both page_size (limit 100) and page for iterate in case that you've more than 100 objects

mandruis7 committed:
Add page_size to the find query

mandruis7 committed:
Update base.py

mandruis7 committed:
Merge pull request #1 from manueliglesiasgarcia/develop

Develop

mandruis7 committed:
Update base.py

mandruis7 committed:
Add page to the find query

So we can iterate over multiple pages due to 100 seems to be the limit of page_size

mandruis7 committed:
Fix lint error

opus-42 commented on supersetapiclient/base.py:
```diff
@@ -258,7 +258,7 @@ def get(self, id: int):
         return object
 
-    def find(self, **kwargs):
+    def find(self, page_size=100, page=0, **kwargs):
```
Can you add type hints here please ?

opus-42 commented on supersetapiclient/base.py:
```diff
@@ -271,13 +271,18 @@ def find(self, **kwargs):
+            query = {"page_size": page_size, "page": page}
+            params = {
+                "q": json.dumps(query)
+            }
```
I think you should factorise that and initialise the query before the if... else (line 266)

```python
       query = {
                "page_size": page_size,
                "page": page
       }

        if kwargs != {}:
        
        ...
        
             query = query.update({...})
             params = ...
```

mandruis7 committed:
Update base.py

To tackle discussion points due to initialice query and as well to remove if ... else due to It's not required anymore

mandruis7 committed:
Update base.py

Fix lint error

mandruis7 commented:
I've updated the MR with some updates with this new code the if ... else It's no longer required and I'm as well initializing the query as you requested

opus-42 commented:
Thank you @nogreintel 😄
